src/test/scala/org/ergoplatform/mining/AutolykosPowSchemeSpec.scala

```diff
@@ -12,12 +13,13 @@ class AutolykosPowSchemeSpec extends ErgoPropertyTest with NoShrink {
     forAll(invalidHeaderGen, Gen.choose(1, 20), Gen.choose[Byte](1, 2)) { (inHeader, difficulty, ver) =>
       val nBits = RequiredDifficulty.encodeCompactBits(difficulty)
       val h = inHeader.copy(nBits = nBits, version = ver)
-      pow.validate(h) shouldBe 'failure
       val sk = randomSecret()
       val x = randomSecret()
       val msg = pow.msgByHeader(h)
       val b = pow.getB(h.nBits)
-      val newHeader = pow.checkNonces(ver, msg, sk, x, b, 0, 1000).map(s => h.copy(powSolution = s)).get
+      val hbs = Ints.toByteArray(h.height)
+      val newHeader = pow.checkNonces(ver, hbs, msg, sk, x, b, 0, 1000)
+        .map(s => h.copy(powSolution = s)).get
       pow.validate(newHeader) shouldBe 'success
     }
   }
```

Comment:
I suggest to use test vectors in addition to generators.
1) Add test vectors in v3.x branch and merge them in master
2) merge them into this v4.0 branch and make sure the v1 version works with them
3) add new test vectors for v2

Reply:
Well, to check that validation procedure is working for v1 blocks, there are "it2" tests already (checking node syncing).